cli/src/complete.rs

```diff
@@ -241,6 +241,7 @@ fn revisions(revisions: Option<&str>) -> Vec<CompletionCandidate> {
         const REMOTE_BOOKMARK: usize = 5;
         const REVSET_ALIAS: usize = 6;
 
+        let match_pattern = (!match_prefix.is_empty()).then(|| format!("glob:{match_prefix}*"));
```

Comment:
maybe better to do `glob::Pattern::escape`?

(I guess glob meta characters wouldn't match the regex pattern, but relying on that would be a source of future bugs.)